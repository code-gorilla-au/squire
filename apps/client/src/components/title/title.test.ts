import { render, screen } from "@testing-library/svelte";
import { describe, expect, it } from "vitest";

import Title from "$components/title/title.test.svelte";

describe("title component", () => {
	it("should render the title", () => {
		render(Title);

		expect(screen.getByRole("heading")).toBeVisible();
	});
	it("should render the back button", () => {
		render(Title);
		expect(screen.getByRole("button")).toBeVisible();
	});
	it('heading should have the text "hello"', () => {
		render(Title);
		expect(screen.getByText("Hello World")).toBeInTheDocument();
	});
});
