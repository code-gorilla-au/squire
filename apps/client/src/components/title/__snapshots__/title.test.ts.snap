// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`title component > should match snapshot 1`] = `"<div class="flex items-center"><button class="bg-muted p-1 flex items-center justify-center rounded-full hover:bg-primary/90 hover:text-primary-foreground"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide-icon lucide lucide-chevron-left"><!----><path d="m15 18-6-6 6-6"></path><!----><!----><!----></svg><!----></button> <h1 class="ml-1 heading-1">Hello World<!----></h1></div>"`;
